Assets/Scripts/Game/Entity/Worker.cs

```diff
@@ -96,6 +102,38 @@ private void StopBuilding()
         state = UnitState.IDLE;
     }
 
+    public void Repair(Building building)
+    {
+        assignedBuilding = building;
+        state = UnitState.REPAIRING;
+    }
+
+    private void UpdateRepair()
+    {
+        if (assignedBuilding == null) return;
+
+        float repairDistSqr = Utils.GetSqrDistance(this, assignedBuilding);
+        if (repairDistSqr < maxRepairDist * maxRepairDist)
+        {
+            bool finished = assignedBuilding.GainHealth(repairRate * Time.deltaTime);
+            if (finished)
+            {
+                StopRepairing();
+            }
```

Comment:
Nit picking, but how about 

```suggestion
            if (finished) StopRepairing();
```
?